onnxruntime/core/providers/rocm/roctracer_manager.h

```diff
@@ -0,0 +1,122 @@
+#pragma once
+
+#include <atomic>
+#include <map>
+#include <memory>
+#include <mutex>
+#include <unordered_map>
+#include <vector>
+
+#include <hip/hip_runtime_api.h>
+#include <roctracer/roctracer.h>
+#include <roctracer/roctracer_hcc.h>
+#include <roctracer/roctracer_hip.h>
+#include <roctracer/roctracer_ext.h>
+#include <roctracer/roctracer_roctx.h>
+
+#include "core/common/profiler_common.h"
+
+namespace onnxruntime {
+namespace profiling {
+
+class RoctracerActivityBuffer {
+ public:
+  RoctracerActivityBuffer()
+      : data_(nullptr), size_(0) {}
+
+  RoctracerActivityBuffer(const uint8_t* data, size_t size)
+      : data_((uint8_t*)malloc(size)), size_(size) {
+    memcpy(data_, data, size);
+  }
+
+  RoctracerActivityBuffer(const RoctracerActivityBuffer& other)
+      : RoctracerActivityBuffer(other.data_, other.size_) {}
+
+  RoctracerActivityBuffer(RoctracerActivityBuffer&& other)
+      : RoctracerActivityBuffer() {
+    std::swap(data_, other.data_);
+    std::swap(size_, other.size_);
+  }
+
+  RoctracerActivityBuffer& operator=(const RoctracerActivityBuffer& other);
+  RoctracerActivityBuffer& operator=(RoctracerActivityBuffer&& other);
+
+  ~RoctracerActivityBuffer();
+
+  // accessors
+  uint8_t* GetData() { return data_; }
+  const uint8_t* GetData() const { return data_; }
+  size_t GetSize() const { return size_; }
+
+ private:
+  uint8_t* data_;
```

Comment:
>uint8_t* data_; [](http://example.com/codeflow?start=2&length=15)

Strongly recommend to be rid the raw free/malloc which might trigger warnings in static analysis ... unique_ptr<char[]> is the option.

Reply:
Thanks, done!